rest/config_legacy.go

```diff
@@ -236,6 +251,62 @@ func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, er
 	return &sc, dbs, nil
 }
 
+func gocbv1tov2AddressUpgrade(server string) (newServer, username, password string, err error) {
+	connSpec, err := connstr.Parse(server)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	splitServers := strings.Split(server, ",")
+	// Only first host is supported to have username and pass or else gocbconnstr.Parse does not work correctly
```

Comment:
The regex used by gocb for part splitting and host splitting doesn't look like it's changed between gocb v1 and v2.  Was gocb v1 doing pre-processing on username/password before using gocbconnstr?

Reply:
Had a look at what Sync Gateway 2.8.2 does and it uses GoLang Parse then fills the dbConfig username and password (if not set) if the user is set in the URL. Will implement this in to here for auto upgrade.